src/Core/Silk.NET.Core/Miscellaneous/RawImage.cs

```diff
@@ -0,0 +1,111 @@
+        public int Width { get; }
+        /// <summary>
+        /// The height of the image in pixels.
+        /// </summary>
+        public int Height { get; }
+        /// <summary>
+        /// The image data.
+        /// </summary>
+        public byte[] Pixels { get; }
+
+        /// <summary>
+        /// Checks whether the two given <see cref="RawImage"/>s are equal.
+        /// </summary>
+        /// <param name="left">The first raw image.</param>
+        /// <param name="right">The second raw image to compare the first against.</param>
+        /// <returns>True if they are equal, false otherwise.</returns>
+        /// <remarks>
+        /// This does not check whether the byte arrays have equal, only whether their references are the same.
+        /// </remarks>  
+        public static bool operator ==(RawImage left, RawImage right) => left.Equals(right);
+
+        /// <summary>
+        /// Checks whether the two given <see cref="RawImage"/>s are not equal.
+        /// </summary>
+        /// <param name="left">The first raw image.</param>
+        /// <param name="right">The second raw image to compare the first against.</param>
+        /// <returns>True if they are not equal, false otherwise.</returns>
+        /// <remarks>
+        /// This does not check whether the byte arrays have equal, only whether their references are the same.
+        /// </remarks>
+        public static bool operator !=(RawImage left, RawImage right) => !(left == right);
+
+        /// <summary>
+        /// Checks whether the given <see cref="RawImage"/> is equal to this one.
+        /// </summary>
+        /// <param name="other">The raw image to compare this raw image against.</param>
+        /// <returns>True if they are equal, false otherwise.</returns>
+        /// <remarks>
+        /// This does not check whether the byte arrays have equal, only whether their references are the same.
+        /// For a complete comparison, use <see cref="ImageEquals"/>
+        /// </remarks>  
+        public bool Equals(RawImage other)
+        {
+            return Width == other.Width && Height == other.Height && Equals(Pixels, other.Pixels);
+        }
+
+        /// <summary>
+        /// Checks whether the given <see cref="RawImage"/> is equal to this one.
+        /// </summary>
+        /// <param name="other">The raw image to compare this raw image against.</param>
+        /// <returns>True if they are equal, false otherwise.</returns>
+        /// <remarks>
+        /// This performs a value comparison over the pixel array, comparing whether the whole image is equal. However,
+        /// this results in the comparison being a lot slower. For just a reference comparison of the pixels array, use
+        /// <see cref="Equals(RawImage)"/>
+        /// </remarks>
+        public bool ImageEquals(RawImage other)
+        {
+            return Width == other.Width && Height == other.Height && Pixels.SequenceEqual(other.Pixels);
+        }
```

Comment:
I don't understand why you chose the methods this way around, I would expect Equals to be sequence equals and maybe another overload to check for references only.

Reply:
On second thought I'm removing ImageEquals because I don't think it's our responsibility to compare the actual image contents. Moreover, this code is not compatible with Memory\<byte\>

Reply:
On second on second thought thought, I'm not doing Memory\<byte\> because we can't pin it!

Reply:
Memory<T>.Span can be pinned.

Reply:
aaaaaaaa ok, so my first comment is my actual comment :P